lib/network/rest/server.js

```diff
@@ -0,0 +1,332 @@
+'use strict'; // eslint-disable-line
+
+const httpServer = require('../http/server');
+const Logger = require('werelogs').Logger;
+const url = require('url');
+const assert = require('assert');
+
+const constants = require('../../constants');
+const utils = require('./utils');
+const errors = require('../../errors');
+
+/**
+ * Parse the Range header into a two-integers array with possibly
+ * undefined values
+ *
+ * @param {String} rangeHeader - The 'Range' header value
+ * @return {(Number|undefined) []} An array of [left, right] integer
+ *   values provided in the range specification if present, or
+ *   undefined otherwise.
+ */
+function parseRangeSpec(rangeHeader) {
```

Comment:
I think it's ideal if we move this piece into Arsenal and make S3 use the same one. We will have one fully tested piece. I have seen quite a few instances in the past where the code is duplicated for simple utilities where there is a bug in one and not the other. One becomes a fully tested piece and the other one not. Also that's the whole purpose of Arsenal. It's meant to have this collection of utilities (big/small) which can be shared across.